src/core/utils/misc.cpp

```diff
@@ -315,30 +315,58 @@ QString Utils::Misc::pythonVersionComplete() {
     return version;
 }
 
+QString Utils::Misc::unitString(Utils::Misc::SizeUnit unit)
+{
+    return QCoreApplication::translate("misc",
+         units[static_cast<int>(unit)].source, units[static_cast<int>(unit)].comment);
+}
+
 // return best userfriendly storage unit (B, KiB, MiB, GiB, TiB)
 // use Binary prefix standards from IEC 60027-2
 // see http://en.wikipedia.org/wiki/Kilobyte
 // value must be given in bytes
 // to send numbers instead of strings with suffixes
-QString Utils::Misc::friendlyUnit(qreal val, bool is_speed)
+
+bool Utils::Misc::friendlyUnit(qreal rawVal, qreal &val, Utils::Misc::SizeUnit &unit)
 {
-    if (val < 0)
-        return QCoreApplication::translate("misc", "Unknown", "Unknown (size)");
+    if (rawVal < 0) {
+        return false;
+    }
     int i = 0;
-    while(val >= 1024. && i < 4) {
-        val /= 1024.;
+    while(rawVal >= 1024. && i < 4) {
```

Comment:
Placing each condition in the parentheses improves the readability of the code, IMO.
```c++
while ((rawVal >= 1024.) && (i < 4))
```